internal/cli/atlas/setup/setup_cmd.go

```diff
@@ -157,14 +162,14 @@ func Builder() *cobra.Command {
 			// registration pre run if applicable
 			if !opts.skipRegister {
 				preRun = append(preRun,
-					opts.register.LoginPreRun(cmd.Context(), config.Default()),
+					opts.register.LoginPreRun(cmd.Context()),
 					validate.NoAPIKeys,
 					validate.NoAccessToken,
 				)
 			}
 
-			if !opts.skipLogin {
-				preRun = append(preRun, opts.register.LoginPreRun(cmd.Context(), config.Default()))
+			if !opts.skipLogin && opts.skipRegister {
+				preRun = append(preRun, opts.register.LoginPreRun(cmd.Context()))
```

Comment:
if opts.skipRegister is true, then preRun already has opts.register.LoginPreRun (see line 165). With this additional check we avoid opts.register.LoginPreRun being run twice